deepcell/image_generators/cropping.py

```diff
@@ -0,0 +1,576 @@
+              (samples, channels, height, width).
+            - It defaults to the image_data_format value found in your
+              Keras config file at "~/.keras/keras.json".
+            - If you never set it, then it will be "channels_last".
+
+        validation_split (float): Fraction of images reserved for validation
+            (strictly between 0 and 1).
+    """
+
+    def __init__(self,
+                 featurewise_center=False,
+                 samplewise_center=False,
+                 featurewise_std_normalization=False,
+                 samplewise_std_normalization=False,
+                 zca_whitening=False,
+                 zca_epsilon=1e-6,
+                 rotation_range=0,
+                 width_shift_range=0.,
+                 height_shift_range=0.,
+                 brightness_range=None,
+                 shear_range=0.,
+                 zoom_range=0.,
+                 channel_shift_range=0.,
+                 fill_mode='nearest',
+                 cval=0.,
+                 horizontal_flip=False,
+                 vertical_flip=False,
+                 rescale=None,
+                 preprocessing_function=None,
+                 data_format='channels_last',
+                 validation_split=0.0,
+                 interpolation_order=1,
+                 crop_size=None,
+                 dtype='float32'):
+
+        ImageDataGenerator.__init__(self,
+                                    featurewise_center=featurewise_center,
+                                    samplewise_center=samplewise_center,
+                                    featurewise_std_normalization=featurewise_std_normalization,
+                                    samplewise_std_normalization=samplewise_std_normalization,
+                                    zca_whitening=zca_whitening,
+                                    zca_epsilon=zca_epsilon,
+                                    rotation_range=rotation_range,
+                                    width_shift_range=width_shift_range,
+                                    height_shift_range=height_shift_range,
+                                    brightness_range=brightness_range,
+                                    shear_range=shear_range,
+                                    zoom_range=zoom_range,
+                                    channel_shift_range=channel_shift_range,
+                                    fill_mode=fill_mode,
+                                    cval=cval,
+                                    horizontal_flip=horizontal_flip,
+                                    vertical_flip=vertical_flip,
+                                    rescale=rescale,
+                                    preprocessing_function=preprocessing_function,
+                                    data_format=data_format,
+                                    validation_split=validation_split,
+                                    dtype=dtype)
```

Comment:
What are you trying to do here? You can call the parent class' init method with:

```
super(CroppingDataGenerator, self).__init__(...)
```

What you currently have does not change the state of the class (I don't think). The `ImageDataGenerator` is not linked to `self`.

Reply:
So the SemanticDataGenerator doesn't have an __init__ defined. However, I need to initialize this self.crop_size parameter for the cropping generator. 

I assumed that I had to call __init__ with the parameters passed in, which is why I did that way. But since we're not actually changing any of the defaults I'm guessing your version would work, which is much simpler

Reply:
Ah, but the `SemanticDataGenerator` inherits an `init` method from its parent class, `ImageDataGenerator`. The way you have it now is not initializing the parent class.

Since you have to override the init method at all, we can just initialize the parent class first, then add your custom attributes. We still need to pass all those default values through though.